transaction/src/builder/manifest_builder_v2.rs

```diff
@@ -0,0 +1,1394 @@
+use crate::internal_prelude::*;
+use radix_engine_interface::api::node_modules::auth::*;
+use radix_engine_interface::api::node_modules::metadata::*;
+use radix_engine_interface::api::node_modules::royalty::*;
+use radix_engine_interface::api::node_modules::ModuleConfig;
+use radix_engine_interface::api::ObjectModuleId;
+use radix_engine_interface::blueprints::access_controller::*;
+use radix_engine_interface::blueprints::account::*;
+use radix_engine_interface::blueprints::consensus_manager::*;
+use radix_engine_interface::blueprints::identity::*;
+use radix_engine_interface::blueprints::package::*;
+use radix_engine_interface::blueprints::resource::ResourceAction::{Burn, Mint};
+use radix_engine_interface::blueprints::resource::*;
+
+/// Utility for building transaction manifest.
+pub struct ManifestBuilderV2 {
```

Comment:
It seems like we now have a lot of duplicate logic between the ManifestBuilderV1 and the ManifestBuilderV2. I think that we can improve the situation there down to something that allows us to maintain both easily. Consider defining a `ManifestBuilder` trait that implements all of the shared logic between V1 and V2 of the manifest builder, something along the lines of:

```rs
pub trait ManifestBuilder {
    pub fn add_instruction(&mut self, instruction: InstructionV1) -> &mut Self;

    pub fn add_blob(&mut self, blob: Vec<u8>) -> &mut Self;

    pub fn assert_worktop_contains(&mut self, resource_address: ResourceAddress, amount: Decimal) -> &mut Self {
        self.add_instruction(InstructionV1::AssertWorktopContains {
            resource_address,
            amount
        })
    }

    /* Other Shared Methods */
}
```

`ManifestBuilderV1` and `ManifestBuilderV2` could then implement this trait and implement `add_instruction` and `add_blob`. Other methods that differ between the two manifest builders could then be implemented in an `impl` block for that specific builder.

Reply:
So chatting with Yulong, I've now replaced ManifestBuilder with this new version.

The ManifestBuilder as-is isn't Toolkit/FFI friendly - it:
* Makes heavy use of traits to make it super-nice to use from Rust, but I doubt translate well over FFI
* Panics if thing aren't correct
* Uses lots of Rust-specific types

I imagine you'll want to instead have a different ManifestBuilder paradigm for the FFI  (much more string-based and less clever). But happy to discuss